runtime/src/bank.rs

```diff
@@ -3752,6 +3768,8 @@ impl Bank {
                     error_counters.account_in_use += 1;
                     Some(index)
                 }
+                Err(TransactionError::WouldExceedMaxBlockCostLimit)
+                | Err(TransactionError::WouldExceedMaxAccountCostLimit) => Some(index),
```

Comment:
Thanks for bring up simulation part. Line #3771/2 add exceeding cost limit as `retry-able` event, alone aide AccountInUse. Caller determines and provides status. 

For simulation, since `prepare_simulation_batch` skips account locking, I didn't add cost-tracking to it either; therefore all transactions passed into `simulate_transaction` will be processed. Eg, above two lines would have no impact. 

In any case, cost_model is deterministic -- it always calculate same transaction to same cost at any time. So if we want to include cost_tracking login into `prepare_simulation_batch`, the result should still be deterministic.

Reply:
Ahh, thanks! I figured you had already thought about simulation; I just needed to consider the different prepare-batch methods.